lib/liquid/block_body.rb

```diff
@@ -93,10 +93,10 @@ def render(context)
         rescue MemoryError => e
           raise e
         rescue UndefinedVariable, UndefinedDropMethod, UndefinedFilter => e
-          context.handle_error(e, token.line_number)
+          context.handle_error(e, token.line_number, token.raw)
           output << nil
         rescue ::StandardError => e
-          output << context.handle_error(e, token.line_number)
+          output << context.handle_error(UnhandledError.new(e), token.line_number, token.raw)
```

Comment:
@wvanbergen Was this kind of what you were thinking, in terms of wrapping the error? My thought was to wrap the error, but still as a different type, so we'd be able to differentiate between error which arise in rendering the liquid vs. something internal that happened in the code which is involved in rendering the liquid cc @dylanahsmith  https://github.com/Shopify/shopify/issues/83649#issuecomment-252432948